pkg/filter/filter.go

```diff
@@ -127,6 +127,13 @@ type ResourceRequirement struct {
 	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
+// DeepCopyInto provides explicit deep copy implementation to avoid
+func (r ResourceRequirement) DeepCopyInto(out *ResourceRequirement) {
```

Comment:
Please also add a simple unit test that confirms that this does do a DeepCopy

Reply:
@tdmanv are we ready to pull the hold off merging?